### ISSUE TITLE
Restrictive Content-Type check in StashClient

### ISSUE THREAD

**Issue by re-schneider** [AmadeusITGroup/sonar-stash#72]
One of the recent change with the StashClient introduced a false-positive.

If the JSON response gets an explicit charset parameter along with the application/json Content-Type, the code will wrongly conclude it is not JSON based on a strict comparison
(see https://www.w3.org/International/articles/http-charset/index).

I created PR #71 with the fix to have it integrated with 1.2.0 release :)

(Thanks to @acopet for the bug report).

**Comment by t-8ch** [comment]
The article specifies (emphasis mine)
```
Documents transmitted with HTTP that are of type *text*, such as text/html, text/plain, etc., can send a charset parameter in the HTTP header to specify the character encoding of the document. 
```
[RFC 4627](https://tools.ietf.org/html/rfc4627#section-6) also specifies:
```
Required parameters: n/a
Optional parameters: n/a
```

**Comment by acopet** [comment]
FYI, my service returns "application/json;charset=UTF-8" instead of only "application/json".
Seems to be relevant to replace **equals** by **contains** method so.

**Comment by re-schneider** [comment]
Integrated with PR #78
